packages/flutter_tools/lib/src/ios/migrations/ios_migrator.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../../base/file_system.dart';
+import '../../base/logger.dart';
+
+/// iOS project is generated from a template on Flutter project creation.
+/// Sometimes (due to behavior changes in Xcode, CocoaPods, etc) these files need to be altered
+/// from the original template.
+class IOSMigrator {
```

Comment:
This would be a bit more idiomatic as an abstract base class
```dart
abstract class IOSMigrator {
  final Logger logger;

  bool migrate();

  String processLine(String line);

  void processFileLines(File file)  {
    // as before.
  }
}

// You can chain them together:
class IOSMigration {
  IOSMigration(this.migrators);

  final List<IOSMigrator> migrators;

  void run() {
    for (final IOSMigrator migrator in migrators) {
      if (!migrator.migrate()) {
        // This class would encapsulate logic for aborting/rolling-back the transaction.
      }
    } 
  }
}
```

Reply:
Followup in https://github.com/flutter/flutter/pull/51879.